crates/ironrdp-session/src/legacy.rs

```diff
@@ -19,19 +21,39 @@ pub fn encode_dvc_message(
         flags: vc::ChannelControlFlags::FLAG_FIRST | vc::ChannelControlFlags::FLAG_LAST,
     };
 
-    // [ TPKT | TPDU | SendDataRequest | vc::ChannelPduHeader | …
-    let written = encode_send_data_request(initiator_id, drdynvc_id, &channel_header, buf).map_err(map_error)?;
-    debug_assert_eq!(written, buf.filled_len());
+    let dvc_message = DvcMessage { channel_header, dvc_pdu, dvc_data };
 
-    // … | dvc::ClientPdu | …
-    dvc_pdu.to_buffer(&mut buf)?;
+    let previous_length = buf.filled_len();
+    // [ TPKT | TPDU | SendDataRequest | vc::ChannelPduHeader | vc::dvc::ClientPdu | DvcData ]
+    let written = encode_send_data_request(initiator_id, drdynvc_id, &dvc_message, buf).map_err(map_error)?;
+    debug_assert_eq!(written, buf.filled_len()-previous_length);
 
-    // … | DvcData ]
-    buf.write_slice(dvc_data);
+    Ok(())
+}
 
-    debug_assert_eq!(buf.filled_len(), written + dvc_length);
+struct DvcMessage<'a> {
+    channel_header: vc::ChannelPduHeader,
+    dvc_pdu: vc::dvc::ClientPdu,
+    dvc_data: &'a[u8],
+}
 
-    Ok(())
+impl PduParsing for DvcMessage<'_> {
+    type Error = ChannelError;
+
+    fn from_buffer(_: impl Read) -> Result<Self, Self::Error> where Self: Sized {
+        todo!()
```

Comment:
suggestion: Just to be on the safe side.
```suggestion
        Err(std::io::Error::other("legacy::DvcMessage::from_buffer called – this is a bug").into())
```